src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -122,6 +124,7 @@ protected ShardIterator shards(ClusterState clusterState, InternalRequest reques
 
         long[] preVersions = new long[request.items().length];
         VersionType[] preVersionTypes = new VersionType[request.items().length];
+        Translog.Location location = null;
         for (int requestIndex = 0; requestIndex < request.items().length; requestIndex++) {
```

Comment:
the point here is that we only sync upto the largest location. A sync guarantees that all smaller locations are synced as well. Does this make sense?